### PR TITLE
[Docs] Improve docs for MMS loading of other languages

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Clarifies #24223 and improves docs of MMS.

### CODE REVIEW

File: docs/source/en/model_doc/mms.mdx [hunk 1]
```diff
@@ -44,11 +44,51 @@ MMS's architecture is based on the Wav2Vec2 model, so one can refer to [Wav2Vec2
...
+
+model_id = "facebook/mms-1b-all"
+target_lang = "fra"
+
+pipe = pipeline(model=model_id, model_kwargs={"target_lang":"fra", "ignore_mismatched_sizes":True})
+```
+
+## Inference
 
+Next, let's look at how we can run MMS in inference and change adapter layers after having called [`~ModelMixin.from_pretrained`]
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/mms.mdx:91]:
```suggestion
Next, let's look at how we can run MMS in inference and change adapter layers after having called [`~PretrainedModel.from_pretrained`]
```

File: docs/source/en/model_doc/mms.mdx [hunk 2]
```diff
@@ -44,11 +44,51 @@ MMS's architecture is based on the Wav2Vec2 model, so one can refer to [Wav2Vec2
 
 The original code can be found [here](https://github.com/facebookresearch/fairseq/tree/main/examples/mms).
 
-## Inference
+## Loading
+
+By default MMS loads adapter weights for English. If you want to load adapter weights of another language 
+make sure to specify `"target_lang=<your-chosen-target-lang>"` as well as `"ignore_mismatched_sizes=True"`.
```
Comment (sgugger) [inline docs/source/en/model_doc/mms.mdx:50]:
```suggestion
make sure to specify `target_lang=<your-chosen-target-lang>` as well as `"gnore_mismatched_sizes=True`.
```
Why the double quotes?

File: docs/source/en/model_doc/mms.mdx [hunk 3]
```diff
@@ -44,11 +44,51 @@ MMS's architecture is based on the Wav2Vec2 model, so one can refer to [Wav2Vec2
...
+The `"ignore_mismatched_sizes=True"` keyword has to be passed to allow the language model head to be resized according
```
Comment (sgugger) [inline docs/source/en/model_doc/mms.mdx:51]: Same
```suggestion
The `ignore_mismatched_sizes=True` keyword has to be passed to allow the language model head to be resized according
```

File: docs/source/en/model_doc/mms.mdx [hunk 4]
```diff
@@ -44,11 +44,51 @@ MMS's architecture is based on the Wav2Vec2 model, so one can refer to [Wav2Vec2
...
+model = Wav2Vec2ForCTC.from_pretrained(model_id, target_lang=target_lang, ignore_mismatched_sizes=True)
+```
 
-By default MMS loads adapter weights for English, but those can be easily switched out for another language.
-Let's look at an example.
+<Tip>
+
+You can safely ignore a warning such as:
+
+```
```
Comment (sgugger) [inline docs/source/en/model_doc/mms.mdx:69]: This should be 
````suggestion
```text
````
or you will have some weird syntax highlighting.

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**